cursorless.nvim/test/nvim-shim.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+if ! [[ "${PWD}" == *"cursorless.nvim" ]]; then
+  echo "ERROR: This script must be run from inside cursorless.nvim/ directory"
+  exit 1
+fi
+
+test_folder=$(mktemp -d "${TMPDIR-/tmp}"/cursorless-busted-test-XXXXX)
+export XDG_CONFIG_HOME="${test_folder}/xdg/config/"
+export XDG_STATE_HOME="${test_folder}/xdg/local/state/"
+export XDG_DATA_HOME="${test_folder}/xdg/local/share/"
+dependency_folder="${XDG_DATA_HOME}/nvim/site/pack/testing/start/"
+plugin_folder="${XDG_CONFIG_HOME}/nvim/plugin/"
+
+mkdir -p "${plugin_folder}" "${XDG_STATE_HOME}" "${dependency_folder}"
+ln -sf "${PWD}" "${dependency_folder}/cursorless.nvim"
+
+# Link in standalone helper functions we want all tests to be able to call
+ln -sf "${PWD}/test/helpers.lua" "${plugin_folder}/helpers.lua"
+
+# shellcheck disable=SC2068
+command nvim --cmd 'set loadplugins' -l $@
```

Comment:
does using `-V1` allow having better output than this?

```
Run cd cursorless.nvim
:noh
:'<,'>noh
:'<,'>noh
+++++
5 successes / 0 failures / 0 errors / 0 pending : 0.002549 seconds
1, 3, 1, 5, true
```

it woud be nice if we could at least have the name of each test printed so we know what is failing once it happens.

See https://neovim.io/doc/user/starting.html#-l

Reply:
I'll look into it. I actually don't get why the output from each test that is there gets printed as is even. Would be nice if the noh stuff could be silenced. 

I guess the most important thing is more what it looks like when one fails so it's easily identifiable. But ya may as well print out each passing test too. I kind of expect busted to do something like that instead of having to print and show with -V1 ourselves, but will see.

Reply:
Yes, I guess it would be nice to know what the output is when a test fails if you can share? I can't test myself.

Reply:
I fixed all the output now, so it's not got the extra cruft. I didn't bother using a custom output handler, I just use the [TAP](https://testanything.org/) one, which is a well defined standard. I also no longer defer output, so it prints each test as ok/not ok as it runs them.

A failed output looks like this:

```
ok 1 - cursorless.nvim tests window_get_visible_lines() Can read one visible line
ok 2 - cursorless.nvim tests window_get_visible_lines() Can read all lines visible on the window
ok 3 - cursorless.nvim tests select_range() Selects the specified range
ok 4 - cursorless.nvim tests buffer_get_selection() Can get the forward selection in a format expected by cursorless
not ok 5 - cursorless.nvim tests buffer_get_selection() Can get the backward selection in a format expected by cursorless
# ./test/unit/cursorless_spec.lua @ 57
# Failure message: ./test/unit/cursorless_spec.lua:63: assertion failed!
1..5
E5113: Error while calling lua chunk: [NULL]
```

The only thing I don't like (minor nit) about the TAP output is the `1..5` at the end, which indicates it ran tests 1 through 5, as it's a bit confusing at first. I thought it was more vim spam lol.

Reply:
I think it is fine like this <3